<launch>
  <!--Launching the Kinect, with most of its features deactivated-->
  <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
    <arg name="rgb_processing" value="true"/><!--ar_pose needs rgb_data-->
    <arg name="depth_processing" value="false"/>
    <arg name="depth_registered_processing" value="false"/>
    <arg name="depth_registration" value="false"/>
    <arg name="disparity_processing" value="true"/>
    <arg name="disparity_registered_processing" value="true"/>
    <arg name="scan_processing" value="false"/>
    <arg name="publish_tf" value="true" /><!--standard false; true if you dont work on the roomba-->
  </include>

  <node name="ar_multi" pkg="ar_pose" type="ar_multi" respawn="false" output="screen">
    <param name="marker_pattern_list" type="string" value="$(find wheeled_robin_formation_drive)/data/objects_kinect"/>
    <param name="use_history" type="bool" value="true"/>
    <param name="threshold" type="int" value="100"/>
    <remap from="/camera/image_raw" to="/camera/rgb/image_raw" />
    <remap from="/camera/camera_info" to="/camera/rgb/camera_info" />
  </node>

  <param name="goal_frame" type="string" value="goal"/>	
  <param name="marker_frame" type="string" value="master_pattern"/>
  <node name="marker_validation" pkg="wheeled_robin_formation_drive" type="marker_validation" output="screen" >
    <param name="x_offset" type="double" value="1.0"/>
    <param name="y_offset" type="double" value="1.0"/>
  </node>
  <node name="calc_vel" pkg="wheeled_robin_formation_drive" type="vel_control" output="screen">
    <param name="p_lin" type="double" value="4.0"/>
    <param name="p_ang" type="double" value="0.5"/>
    <remap from="/cmd_vel" to="/cmd_vel_mux/input/teleop" />
  </node>

<include file="$(find turtlebot_navigation)/launch/includes/move_base.launch.xml"/>


</launch>
